pkg/upgrade/preflight/config_check_test.go

```diff
@@ -0,0 +1,266 @@
+/*
+Copyright 2023 The Radius Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package preflight
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/radius-project/radius/pkg/cli/filesystem"
+	"github.com/radius-project/radius/pkg/cli/helm"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+)
+
+func TestCustomConfigValidationCheck_BasicValidation(t *testing.T) {
+	tests := []struct {
+		name              string
+		setParams         []string
+		setFileParams     []string
+		setupFiles        func(t *testing.T) []string
+		shouldFail        bool
+		expectMsgContains []string
+	}{
+		{
+			name: "no parameters",
+			expectMsgContains: []string{
+				"No custom configuration parameters provided",
+			},
+		},
+		{
+			name: "valid parameters - mixed types",
+			setParams: []string{
+				"app.config=value",
+				"image.tag=v1.0.0",
+			},
+			setupFiles: func(t *testing.T) []string {
+				tmpFile := createTempFile(t, "test content")
+				return []string{fmt.Sprintf("values.yaml=%s", tmpFile)}
+			},
+			expectMsgContains: []string{
+				"All 3 custom configuration parameters passed",
+				"basic validation",
+			},
+		},
+		{
+			name: "invalid parameters - mixed failures",
+			setParams: []string{
+				"valid.key=value",
+				"invalid-format",
+				"=empty-key",
+			},
+			setFileParams: []string{
+				"config=/nonexistent/file.yaml",
+				"=empty-key-file",
+			},
+			shouldFail: true,
+			expectMsgContains: []string{
+				"Configuration validation failed",
+				"must be in format 'key=value'",
+				"key cannot be empty",
+				"cannot read file",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			setFileParams := tt.setFileParams
+			if tt.setupFiles != nil {
+				setFileParams = tt.setupFiles(t)
+			}
+
+			check := NewCustomConfigValidationCheck(tt.setParams, setFileParams, "", nil)
+			// Override the default chart path to skip chart validation for basic tests
+			check.chartPath = ""
+			pass, msg, err := check.Run(context.Background())
+
+			require.NoError(t, err)
+			assert.Equal(t, !tt.shouldFail, pass)
+			for _, contains := range tt.expectMsgContains {
+				assert.Contains(t, msg, contains)
+			}
+		})
+	}
+}
+
+func TestCustomConfigValidationCheck_ChartValidation(t *testing.T) {
+	// Skip if chart doesn't exist
+	if _, err := os.Stat("../../../deploy/Chart"); os.IsNotExist(err) {
```

Comment:
can we use memmapfilesystem for this? using `os` commands in unit tests probably isn't a good idea